TrenDAP/wwwroot/TypeScript/Features/DataSources/EditDataSource.tsx

```diff
@@ -27,40 +27,42 @@ import { DataSourceTypes } from '../../global';
 import { useAppDispatch } from '../../hooks';
 import { UpdateDataSource } from './DataSourcesSlice'
 import DataSource from './DataSource';
-import { Pencil } from './../../Constants'
-const EditDataSource: React.FunctionComponent<{ DataSource: DataSourceTypes.IDataSourceView}> = (props) => {
+import { Modal } from '@gpa-gemstone/react-interactive';
+import { CrossMark, Pencil } from '@gpa-gemstone/gpa-symbols';
+
+const EditDataSource: React.FunctionComponent<{ DataSource: DataSourceTypes.IDataSourceView }> = (props) => {
     const dispatch = useAppDispatch();
     const [dataSource, setDataSource] = React.useState<DataSourceTypes.IDataSourceView>(props.DataSource);
     const [show, setShow] = React.useState<boolean>(false);
+    const [errors, setErrors] = React.useState<string[]>([]);
+
+    React.useEffect(() => {
+        const e = [];
+        if (dataSource.Name == null || dataSource.Name.trim().length == 0)
+            e.push("A Name has to be entered.")
```

Comment:
should `Name` be unique? @elwills thoughts?

Reply:
@prestoncraw had a discussion with Erika and Gabriel. Can you add the logic to show an error if that name already exists for a shared dataset or a dataset for that user?. UI logic only since we don't want to look through all datasets, only those the user has access to

Reply:
Here is that logic implemented for data sets
https://github.com/GridProtectionAlliance/TrenDAP/commit/422678f1e6d105e0ba51ebe0d4aafc430d4b5375
See also this ticket on why some of this logic may be necessary
https://gridprotectionalliance.atlassian.net/browse/TDAP-30